### PR TITLE
fix NPE when updatePublishDispatcher

### PR DESCRIPTION
**Pull request by lightzhao**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

When updatePublishDispatcher, it is not judged whether the policies is empty, which will cause NPE.

### Modifications

fix NPE

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by lightzhao** [comment]
@eolivelli   Please help me review, thank you

**Comment by lightzhao** [comment]
hello, @codelipenghui Please help me review and merge, thank you.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -883,7 +883,7 @@ private void updatePublishDispatcher(Optional<Policies> optPolicies) {
 
         //topic-level policy is not set, try to use namespace-level rate policy
         final String clusterName = brokerService.pulsar().getConfiguration().getClusterName();
-        final PublishRate publishRate = policies.publishMaxMessageRate != null
+        final PublishRate publishRate = policies !=null && policies.publishMaxMessageRate != null
```
Comment (Jason918) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:886]: Why would this policies be null?  Do you have any case?

Comment (lightzhao) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:886]: There is no NPE case in the production environment,But I see that the logic of the code below is judged to be null.
[// attach the resource-group level rate limiters, if set
        String rgName = policies != null && policies.resource_group_name != null
          ? policies.resource_group_name
          : null;]

**Comment by mattisonchao** [comment]
@lightzhao  

I think we don't need to fix this.

You can see the obvious hint in here.( In IDEA )

image


**NOTICE: Maybe you can help Pulsar clean this code.**

**Comment by lightzhao** [comment]
> @lightzhao @Jason918
> 
> I think we don't need to fix this.
> 
> You can see the obvious hint in here.( In IDEA )
> 
> image
> 
> **NOTICE: Maybe you can help Pulsar clean this code.**

ok. thanks.

**Pull request closed without merging**